### PR TITLE
[WIP] Feature/port seteron magic

### LINKED ISSUE #1343: Update lighting
## Summary

Update the lighting based on our old lighting from the "outdated" branch. 

## Goal 

Port/remake the lighting in the project to as close as the "outdated" branch as possible.

Original PRs:
- #283 
- #857

### PR DESCRIPTION
**Pull request by joaoburatto**
# Summary

Ports Seteron's lighting and visual updates from the outdated branch.

This doesn't configure or enable SSRT by default, as its a weird post processing, has a bunch of artifacts and might be outdated by current day standards. I instead left AO enabled on the post processing stuff.

Updates the lights to be more appropriate with the new deferred rendering and material updates, also changes the scene lighting to full darkness contrary to full brightness as it was before.

Also moved the global post processing layer to a separate gameobject child to the camera for organization sake.

Fixes the ministation map to work with the new system.

#### PR checklist
- [ ] The game builds properly without errors.
- [ ] No unrelated changes are present.
- [ ] No "trash" files are committed.

# Testing

- [ ] Lighting looks like Seteron's version.
- [ ] Post processing looks like Seteron's version.
- [ ] Performance increases due to Seteron's optimizations. 

# Changes

Changed the item icon shader to unlit to make it work with the deferred rendering.

# Related issues/PRs 

Closes #1343.

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
Did you add gitbook documentation? When you do, link it in the description.

File: Assets/Art/Shaders.meta
Comment (cosmiccoincidence) [inline Assets/Art/Shaders.meta:1]: Shaders are not art assets by our definition. Move this folder to Content/worldObjects/World/VFX/.

Comment (joaoburatto) [reply Assets/Art/Shaders.meta:1]: That classification doesn't make much sense to me. Content/Shaders makes more sense to me.

**Comment by joaoburatto** [comment]
> Did you add gitbook documentation? When you do, link it in the description.

What do you want to add to the GitBook?

**Comment by cosmiccoincidence** [comment]
> > Did you add gitbook documentation? When you do, link it in the description.
> 
> What do you want to add to the GitBook?

Should add a page in Design explaining what the the lighting should look like and why. (soft lighting so no hard shadows, etc)
And a page in systems explaining anything important regarding the implementation.

File: Assets/Art/Shaders.meta
Comment (cosmiccoincidence) [reply Assets/Art/Shaders.meta:1]: Shaders effect WorldObject, a type of VFX in my opinion.

**Comment by stilnat** [comment]
I'm adding @Seteron remarks from discord here, so another reviewer can check if those are correct : 

- The player camera's post processing volume got deleted, meaning it's recieving no post-processing
- The old stuff was all designed for Linear color space but the current project is on gamma, so it doesn't work right
- The old branch was using the half-toon shader I think, not the full one.
- The cgincs aren't working correctly
- The light fixture values aren't correct
- The player's skin is using some kind of emissive shader?
- Ambient lighting is enabled, so everything is way too bright

Here is a reference for around what it should look like with correct settings

image

With shadows:

image

File: Assets/Art/Shaders.meta
Comment (joaoburatto) [reply Assets/Art/Shaders.meta:1]: That's not always true, shaders can be used for UI interactions too

**Comment by joaoburatto** [comment]
On halt due to other priorities

**Pull request closed without merging**